modular_zubbers/code/game/gamemodes/fleshmind/fleshmind_mobs.dm

```diff
@@ -0,0 +1,1504 @@
+	icon_state = "phaser-1"
+	base_icon_state = "phaser"
+	duration = 30 SECONDS
+	/// The target we move towards, if any.
+	var/datum/weakref/target_ref
+
+/obj/effect/temp_visual/phaser/Initialize(mapload, atom/movable/target)
+	. = ..()
+	icon_state = "[base_icon_state]-[rand(1, 3)]"
+	filters += filter(type = "blur", size = 0)
+
+/obj/effect/temp_visual/phaser/proc/parent_phase_move(datum/source, turf/target_atom, nearby)
+	SIGNAL_HANDLER
+	if(!target_atom)
+		return
+	phase_move_to(target_atom, TRUE)
+
+/obj/effect/temp_visual/phaser/proc/parent_death(mob/living/dead_guy, gibbed)
+	SIGNAL_HANDLER
+	qdel(src)
+
+/obj/effect/temp_visual/phaser/proc/phase_move_to(atom/target_atom, nearby = FALSE)
+	var/turf/new_place
+	var/distance_to_target = get_dist(src, target_atom)
+	var/turf/target_turf = get_turf(target_atom)
+	//if our target is near, we move precisely to it
+	if(distance_to_target <= 3)
+		if(nearby)
+			for(var/dir in GLOB.alldirs)
+				var/turf/nearby_turf = get_step(new_place, dir)
+				if(can_jump_on(nearby_turf, target_turf))
+					new_place = nearby_turf
+		else
+			new_place = target_turf
+
+	if(!new_place)
+		//there we make some kind of, you know, that creepy zig-zag moving
+		//we just take angle, distort it a bit and turn into dir
+		var/angle = get_angle(loc, target_turf)
+		angle += rand(5, 25) * pick(-1, 1)
+		if(angle < 0)
+			angle = 360 + angle
+		if(angle > 360)
+			angle = 360 - angle
+		var/tp_direction = angle2dir(angle)
+		new_place = get_ranged_target_turf(loc, tp_direction, rand(2, 4))
+
+	if(!can_jump_on(new_place, target_turf))
+		return
+	//an animation
+	var/init_px = pixel_x
+	animate(src, pixel_x = init_px + 16 * pick(-1, 1), time=5)
+	animate(pixel_x = init_px, time = 6, easing = SINE_EASING)
+	animate(filters[1], size = 5, time = 5, flags = ANIMATION_PARALLEL)
+	addtimer(CALLBACK(src, PROC_REF(phase_jump), new_place), 0.5 SECONDS)
+
+/obj/effect/temp_visual/phaser/proc/phase_jump(turf/target_turf)
+	playsound(target_turf, 'sound/effects/phasein.ogg', 60, 1)
+	animate(filters[1], size = 0, time = 5)
+	icon_state = "[base_icon_state]-[rand(1, 4)]"
+	forceMove(target_turf)
+
+/obj/effect/temp_visual/phaser/proc/can_jump_on(turf/target_turf, turf/previous_turf)
+	if(!target_turf || target_turf.density || isopenspaceturf(target_turf))
+		return FALSE
+
+	if(previous_turf)
+		if(!can_see(target_turf, previous_turf, DEFAULT_VIEW_RANGE))
+			return FALSE
+
+	//to prevent reflection's stacking
+	var/obj/effect/temp_visual/phaser/phaser_reflection = locate() in target_turf
+	if(phaser_reflection)
+		return FALSE
+
+	for(var/obj/iterating_object in target_turf)
+		if(!iterating_object.CanPass(src, target_turf))
+			return FALSE
+
+	return TRUE
+
+
+/**
+ * Mechiver
+ *
+ * Special abilities: Can grab someone and shove them inside, does DOT and flavour text, can convert dead corpses into living ones that work for the flesh.
+ *
+ *
+ */
+/mob/living/basic/fleshmind/mechiver
```

Comment:
A 450 health mob that deals 20-30 damage. The movespeed makes this semi-balanced, but it could have less damage.